src/webgpu/shader/validation/uniformity/uniformity.spec.ts

```diff
@@ -354,3 +355,867 @@ fn main(@builtin(${t.params.builtin}) p : ${t.params.type}) {
+  },
+  unreachable_nonuniform: {
+    typename: `u32`,
+    typedecl: ``,
+    assignment: `loop {
+      break;
+      x = nonuniform_value[0];
+    }`,
+    cond: `x > 0`,
+    uniform: `init`,
+  },
+  if_no_else_uniform: {
+    typename: `u32`,
+    typedecl: ``,
+    assignment: `if uniform_cond {
+      x = uniform_value[0];
+    }`,
+    cond: `x > 0`,
+    uniform: `init`,
+  },
+  if_no_else_nonuniform: {
+    typename: `u32`,
+    typedecl: ``,
+    assignment: `if uniform_cond {
+      x = nonuniform_value[0];
+    }`,
+    cond: `x > 0`,
+    uniform: `never`,
+  },
+  if_no_then_uniform: {
+    typename: `u32`,
+    typedecl: ``,
+    assignment: `if uniform_cond {
+    } else {
+      x = uniform_value[0];
+    }`,
+    cond: `x > 0`,
+    uniform: `init`,
+  },
+  if_no_then_nonuniform: {
+    typename: `u32`,
+    typedecl: ``,
+    assignment: `if uniform_cond {
+    } else {
+      x = nonuniform_value[0];
+    }`,
+    cond: `x > 0`,
+    uniform: `never`,
+  },
+  if_else_uniform: {
+    typename: `u32`,
+    typedecl: ``,
+    assignment: `if uniform_cond {
+      x = uniform_value[0];
+    } else {
+      x = uniform_value[1];
+    }`,
+    cond: `x > 0`,
+    uniform: `always`,
+  },
+  if_else_nonuniform: {
+    typename: `u32`,
+    typedecl: ``,
+    assignment: `if uniform_cond {
+      x = nonuniform_value[0];
+    } else {
+      x = nonuniform_value[1];
+    }`,
+    cond: `x > 0`,
+    uniform: `never`,
+  },
+  if_else_split: {
+    typename: `u32`,
+    typedecl: ``,
+    assignment: `if uniform_cond {
+      x = uniform_value[0];
+    } else {
+      x = nonuniform_value[0];
+    }`,
+    cond: `x > 0`,
+    uniform: `never`,
+  },
+  if_unreachable_else_none: {
+    typename: `u32`,
+    typedecl: ``,
+    assignment: `if uniform_cond {
+    } else {
+      return;
+    }`,
+    cond: `x > 0`,
+    uniform: `init`,
+  },
+  if_unreachable_else_uniform: {
+    typename: `u32`,
+    typedecl: ``,
+    assignment: `if uniform_cond {
+      x = uniform_value[0];
+    } else {
+      return;
+    }`,
+    cond: `x > 0`,
+    uniform: `always`,
+  },
+  if_unreachable_else_nonuniform: {
+    typename: `u32`,
+    typedecl: ``,
+    assignment: `if uniform_cond {
+      x = nonuniform_value[0];
+    } else {
+      return;
+    }`,
+    cond: `x > 0`,
+    uniform: `never`,
+  },
+  if_unreachable_then_none: {
+    typename: `u32`,
+    typedecl: ``,
+    assignment: `if uniform_cond {
+      return;
+    }`,
+    cond: `x > 0`,
+    uniform: `init`,
+  },
+  if_unreachable_then_uniform: {
+    typename: `u32`,
+    typedecl: ``,
+    assignment: `if uniform_cond {
+      return;
+    } else {
+      x = uniform_value[0];
+    }`,
+    cond: `x > 0`,
+    uniform: `always`,
+  },
+  if_unreachable_then_nonuniform: {
+    typename: `u32`,
+    typedecl: ``,
+    assignment: `if uniform_cond {
+      return;
+    } else {
+      x = nonuniform_value[0];
+    }`,
+    cond: `x > 0`,
+    uniform: `never`,
+  },
```

Comment:
Suggest maybe adding a case that does something like this (if not already covered somewhere that I missed):
```
if uniform_cond {
  x = nonuniform_value[0];
  return;
}
```
This tests that the non-uniform assignment doesn't make it out of the if-else because of the return statement, so `x` would still be uniform afterwards.

Reply:
Added in the latest commit.

Reply:
Thanks!